CGE_equations.py

```diff
@@ -0,0 +1,688 @@
+
+    Args:
+        alpha (1D numpy array): Budget share of good i
+        pf (1D numpy array): The price of factor h
+        Ff (1D numpy array): Endowment of factor h
+        Sp (float): Total household savings
+        Td (float): Total direct tax revenue
+        Fsh = Repatriated profits
+        Trf (float): Total transfers to households
+        pq (1D numpy array): price of XXXX for each good i
+
+    Returns:
+        Xp (1D numpy array): Demand for production good i by consumers
+    '''
+    Xp = alpha * ((pf * Ff).sum() - Sp - Td - Fsh + Trf) / pq
+    return Xp
+
+
+def eqpe(er, pWe):
+    '''
+    Export prices.
+
+    .. math::
+        pe_{i} = \varepsilon \cdot pWe_{i}
+
+    Args:
+        er (float): The real exchange rate
+        pWe (1D numpy array): The world price of commodity i in foreign currency
+
+    Returns:
+        pe (1D numpy array): Price of commodity i exports in domestic currency
+    '''
+    pe = er * pWe
+    return pe
+
+
+def eqpm(er, pWm):
+    '''
+    Import prices.
+
+    .. math::
+        pm_{i} = \varepsilon \cdot pWm_{i}
+
+    Args:
+        er (float): The real exchange rate
+        pWm (1D numpy array): The world price of commodity i in foreign currency.
+
+    Returns:
+        pm (1D numpy array): The price of commodity i imports in domestic currency.
+    '''
+    pm = er * pWm
+    return pm
+
+
+def eqbop(pWe, pWm, E, M, Sf, Fsh, er):
+    '''
+    Balance of payments.
+
+    .. math::
+        \sum_{i}pWe_{i}E_{i} + \frac{Sf}{\varepsilon} = \sum_{i}pWm_{i}M_{i} + \frac{Fsh}{\varepsilon}
+
+    Args:
+        pWe (1D numpy array): The world price of commodity i in foreign currency
+        pWm (1D numpy array): The world price of commodity i in foreign currency.
+        E (1D numpy array): Exports of commodity i
+        M (1D numpy array): Imports of commodity i
+        Sf (float): Total foreign savings (??)
+        Fsh = Repatriated profits
+        er (float): The real exchange rate
+
+    Returns:
+        bop_error (float): Error in balance of payments equation.
+
+    '''
+    bop_error = (pWe * E).sum() + Sf / er - ((pWm * M).sum() + Fsh / er)
+    return bop_error
+
+
+def eqSf(g, lam, pq, Kf):
+    '''
+    Net foreign investment/savings.
+
+    .. math::
+        Sf = g Kf \sum_{j} \lambda_{j} pq_{j}
+
+    Args:
+        g (float): Exogenous long run growth rate of the economy
+        lam (1D numpy array): Fixed shares of investment for each good j
+        pq (1D numpy array): price of XXXX for each good i
+        Kf (float): Foreign owned capital ??
+
+    Returns:
+        Sf (float): Total foreign savings (??)
+    '''
+    Sf = g * Kf * (lam * pq).sum()
+    return Sf
+
+
+def eqQ(gamma, deltam, deltad, eta, M, D):
+    '''
+    CES production function for the importing firm.
+
+    .. math::
+        Q_{i} = \gamma_{i}\left[\delta^{m}_{i}M^{\eta_{i}}_{i} + \delta^{d}_{i}D^{\eta_{i}}_{i}\right]^{\frac{1}{\eta_{i}}}
+
+    Args:
+        gamma (1D numpy array): ??
```

Comment:
@DPrihardini What do we call the gamma parameter?

Reply:
It is a scale parameter so that the CES equation can match the data for total supply Q that is reported in the Social Accounting Matrix. Without this, we could not match the data, unless eta is equal to zero.